src/sentry/rules/processing/delayed_processing.py

```diff
@@ -142,84 +149,183 @@ def get_condition_query_groups(
     """
     condition_groups: dict[UniqueConditionQuery, DataAndGroups] = {}
     for rule in alert_rules:
-        # We only want a rule's slow conditions because alert_rules are only added
-        # to the buffer if we've already checked their fast conditions.
         slow_conditions = get_slow_conditions(rule)
         for condition_data in slow_conditions:
             for condition_query in generate_unique_queries(condition_data, rule.environment_id):
-                # NOTE: If percent and count comparison conditions are sharing
-                # the same UniqueConditionQuery, the condition JSON in
-                # DataAndGroups will be incorrect for one of those types.
-                # The JSON will either have or be missing a comparisonInterval
-                # which only applies to percent conditions, and have the incorrect
-                # comparisonType for one type. This is not a concern because
-                # when we instantiate the exact condition class with the JSON,
-                # the class ignores both fields when calling get_rate_bulk.
-
-                # Add to set of group_ids if there are already group_ids
-                # that apply to the unique condition query.
```

Comment:
Without this comment I think people could get tripped up on this in the future

Reply:
👍 agree - my bad, the comments got destroyed by an editor plugin when i tried to do some automation to the file 😅 -- lemme know if you notice any other ones you think are important as well. (i tried to pull back as many as i could 😅)